### PR TITLE
Add lima.yaml as an alternative to Vagrantfile

### LINKED ISSUE #20: Allow to use Lima instead of Vagrant
The blog post mentions using `vagrant`:

https://kubernetes.io/blog/2023/10/10/cri-o-community-package-infrastructure/

But it could be nice to use Lima instead ?

https://lima-vm.io/

### PR DESCRIPTION
**Pull request by afbjorklund**
#### What type of PR is this?

/kind feature

#### Which issue(s) this PR fixes:

Fixes #20

#### Special notes for your reviewer:

https://lima-vm.io has notes on how to install Lima

#### Does this PR introduce a user-facing change?

```release-note
add yaml templates for running test with lima
```

### CODE REVIEW

File: test/deb/lima.yaml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+provision:
+- mode: system
+  script: |
+    #!/bin/bash
+    set -eux -o pipefail
+    test -e /etc/kubernetes/admin.conf && exit 0
+
+      # Package configs
+      . /tmp/lima/env ||:
+      KUBERNETES_VERSION=${KUBERNETES_VERSION:-v1.28}
```
Comment (saschagrunert) [inline test/deb/lima.yaml:26]: We need to change dependencies.yaml for this: https://github.com/cri-o/packaging/blob/c9e58da106e99dce6f3ec9835b4ae2a422137d78/dependencies.yaml#L3

File: test/rpm/lima.yaml [hunk 2]
```diff
@@ -0,0 +1,88 @@
+# This template requires Lima v0.7.0 or later.
+images:
+- location: "https://download.fedoraproject.org/pub/fedora/linux/releases/38/Cloud/x86_64/images/Fedora-Cloud-Base-38-1.6.x86_64.qcow2"
```
Comment (saschagrunert) [inline test/rpm/lima.yaml:3]: Let’s add this to dependencies.yaml.

File: test/deb/lima.yaml [hunk 3]
```diff
@@ -0,0 +1,69 @@
+# This template requires Lima v0.7.0 or later.
+images:
+- location: "https://cloud-images.ubuntu.com/releases/22.04/release/ubuntu-22.04-server-cloudimg-amd64.img"
```
Comment (saschagrunert) [inline test/deb/lima.yaml:3]: Let’s add those to dependencies.yaml

File: test/rpm/lima.yaml [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+provision:
+- mode: system
+  script: |
+    #!/bin/bash
+    set -eux -o pipefail
+    test -e /etc/kubernetes/admin.conf && exit 0
+
+      # Package configs
+      . /tmp/lima/env ||:
+      KUBERNETES_VERSION=${KUBERNETES_VERSION:-v1.28}
```
Comment (saschagrunert) [inline test/rpm/lima.yaml:28]: This needs to be part of dependencies.yaml, too.

**Comment by afbjorklund** [comment]
The lima base templates come from here:

https://github.com/lima-vm/lima/blob/master/examples/ubuntu-lts.yaml

https://github.com/lima-vm/lima/blob/master/examples/fedora.yaml

I removed the home directory (`~`) mount.

**Review by saschagrunert — APPROVED** [review]
/lgtm

**Pull request merged**